module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -54,13 +55,23 @@ func (k Keeper) Handle(ctx sdk.Context, eve types.EthereumEvent) (err error) {
 		return nil
 
 	case *types.BatchExecutedEvent:
-		k.batchTxExecuted(ctx, common.HexToAddress(event.TokenContract), event.BatchNonce)
+		if err := k.batchTxExecuted(ctx, common.HexToAddress(event.TokenContract), event.BatchNonce); err != nil {
+			return err
+		}
 		k.AfterBatchExecutedEvent(ctx, *event)
 		return nil
 
 	case *types.ERC20DeployedEvent:
 		if err := k.verifyERC20DeployedEvent(ctx, event); err != nil {
-			return err
+			// log the error and return nil, otherwise the bridge will be desactivated
```

Comment:
why not to deactivate the bridge? is it for the initial deployment?

Reply:
oh the caller behaviour changed -- previously it just logged

Reply:
Anyone can call the "deployERC20" on the gravity contract with some wrong parameters. The error here should not turn off the bridge